### PR TITLE
Piridsystem: New ID System for WPM

### PR DESCRIPTION
**Pull request by decemberWP**
## Type of change

- [x] Feature

## Description of change

This PR introduces new User ID System for WPM

Module Name: pirIDSystem
Module Type: UserID Module
Maintainer: pawel.grudzien@grupawp.pl

Integration example: 
```
pbjs.setConfig({
    userSync: {
        userIds: [{
            name: 'pirId',
            storage: {
                type: 'cookie',
                name: 'pirIdToken',
                expires: 7,
                refreshInSeconds: 360
            },
        }]
    }
});

### CODE REVIEW

**Comment by ChrisHuie** [comment]
@decemberWP this looks like it is a month or so behind the current master. Pulling in recent commits should fix your errors here

**Comment by decemberWP** [comment]
@ChrisHuie thanks, done

File: modules/pirIdSystem.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  /**
+   * Extracts the top-level domain from the current window's location.
+   *
+   * @returns {string} The top-level domain of the current window's location.
+   */
+  domainOverride() {
+    const topDomain = window.location.hostname.split('.').slice(-2).join('.');
+
+    return topDomain;
```
[the remark this answers is not in the document]
Comment (decemberWP) [reply modules/pirIdSystem.js]: thanks, fixed

File: modules/pirIdSystem.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+   * @returns {(Object|undefined)}
+   */
+  decode(value) {
+    return typeof value === 'string' ? { 'pirId': value } : undefined;
+  },
+
+  /**
+   * performs action to obtain id and return a value
+   * @function
+   * @returns {Id: string|undefined}
```
[the remark this answers is not in the document]
Comment (decemberWP) [reply modules/pirIdSystem.js]: thanks, fixed

File: modules/pirIdSystem.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * This module adds pirId to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/pirId
+ * @requires module:modules/userId
+ */
+
+import { MODULE_TYPE_UID } from '../src/activities/modules.js';
+import { getStorageManager } from '../src/storageManager.js';
+import { submodule } from '../src/hook.js';
+
```
[the remark this answers is not in the document]
Comment (decemberWP) [reply modules/pirIdSystem.js:17]: thanks, fixed

**Review by osazos — APPROVED** [review]
LGTM

**Pull request merged**